sky/backends/backend_utils.py

```diff
@@ -1667,16 +1601,8 @@ def _query_status_aws(
         'terminated': None,
     }
     region = ray_config['provider']['region']
-    launch_hashes = _ray_launch_hash(cluster, ray_config)
-    if launch_hashes is not None:
-        hash_filter_str = ','.join(launch_hashes)
-        hash_filter_line = (
-            f'Name=tag:ray-launch-config,Values={hash_filter_str} ')
-    else:
-        hash_filter_line = ''
     query_cmd = ('aws ec2 describe-instances --filters '
                  f'Name=tag:ray-cluster-name,Values={cluster} '
-                 f'{hash_filter_line}'
```

Comment:
On a quick look, we may want to ensure the following situation still works:

- (Completely normal case) Users who have a cluster in console (up/stopped) and there's no leaked version
- (Normal? case) Users who have a cluster in console (up/stopped) and there are leaked (stopped) versions
  - Can we handle this automatically? If not, then we may need to have some safeguards and ask users to manually clean up.

We probably don't need to handle the adversarial case where users manually create another cluster with the same `tag:ray-cluster-name`.

Reply:
Added the tests in the description. We will currently raise an error if the leakage is detected, as shown above. PTAL. : )